### PR TITLE
satellite/repair: dont error on deleted segments

### PR DESCRIPTION
**Pull request by littleskunk**
What: Clean up the satellite log. Remove the following repair error.
```
{"L":"\u001b[31mERROR\u001b[0m","T":"2019-10-12T11:19:56.227+0200","N":"repairer","C":"repairer/repairer.go:95","M":"repair worker failed:","error":"repairer error: repairing injured segment: repairer error: metainfo error: key not found: <segment path>,"errorVerbose":"repairer error: repairing injured segment: repairer error: metainfo error: key not found: <segment path>"\n\tstorj.io/storj/satellite/repair/repairer.(*Service).worker:124\n\tstorj.io/storj/satellite/repair/repairer.(*Service).process.func1:93\n\tstorj.io/storj/internal/sync2.(*Limiter).Go.func1:41","S":"storj.io/storj/satellite/repair/repairer.(*Service).process.func1\n\t/home/storj/source/satellite/repair/repairer/repairer.go:95\nstorj.io/storj/internal/sync2.(*Limiter).Go.func1\n\t/home/storj/source/internal/sync2/limiter.go:41"}
```

Why: A segment can be deleted between putting it on the repair queue and repairing it. I wouldn't call that an error. Removing it will help us to have a better overview over the remaining repair errors.

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/repair/repairer/segments.go [hunk 1]
```diff
@@ -78,7 +78,9 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (s
 	// Read the segment pointer from the metainfo
 	pointer, err := repairer.metainfo.Get(ctx, path)
 	if err != nil {
```
Comment (jtolds) [inline satellite/repair/repairer/segments.go]: we need to actually confirm that the error is specifically a not found error. some sort of database connection error here doesn't mean the repair is unnecessary

Comment (jtolds) [reply satellite/repair/repairer/segments.go]: there are some other places in the code where we check if the error is NotFound if that helps

Comment (navillasa) [reply satellite/repair/repairer/segments.go]: I think it would be `if storage.ErrKeyNotFound.Has(err)`

Comment (littleskunk) [reply satellite/repair/repairer/segments.go]: Good catch and thank you.

File: satellite/repair/repairer/segments.go [hunk 2]
```diff
@@ -78,7 +78,9 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (s
...
-		return storj.ErrObjectNotFound.Has(err), Error.Wrap(err)
+		mon.Meter("repair_unnecessary").Mark(1)
+		repairer.log.Sugar().Debugf("segment %v was deleted", path)
```
[the remark this answers is not in the document]
Comment (littleskunk) [reply satellite/repair/repairer/segments.go]: Let me open another PR for that. I would like to change the format for the segment path. At the moment it is hard to grep it in the logfile.

File: satellite/repair/repairer/segments.go [hunk 3]
```diff
@@ -78,7 +79,12 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (s
 	// Read the segment pointer from the metainfo
 	pointer, err := repairer.metainfo.Get(ctx, path)
 	if err != nil {
-		return storj.ErrObjectNotFound.Has(err), Error.Wrap(err)
+		if storage.ErrKeyNotFound.Has(err) {
```
Comment (VinozzZ) [inline satellite/repair/repairer/segments.go]: Maybe we should add test case for this too

Comment (littleskunk) [reply satellite/repair/repairer/segments.go]: Challenge accepted.

**Comment by littleskunk** [comment]
Now inclusive unit test :)

File: satellite/repair/repair_test.go [hunk 4]
```diff
@@ -409,6 +409,88 @@ func TestCorruptDataRepair_Succeed(t *testing.T) {
...
+		require.NoError(t, err)
+		require.Equal(t, count, 1)
+
+		// Run the repairer
+		satellite.Repair.Repairer.Loop.Restart()
+		satellite.Repair.Repairer.Loop.TriggerWait()
+		satellite.Repair.Repairer.Loop.Pause()
+		satellite.Repair.Repairer.Limiter.Wait()
+
+		// Verify that the segnebt was removed
```
Comment (VinozzZ) [inline satellite/repair/repair_test.go]: nit - `segnebt` -> `segment`

File: satellite/repair/repairer/segments.go (hunk 3 above)
Comment (egonelbre) [inline satellite/repair/repairer/segments.go]: I wonder whether the correct error should be `ErrObjectNotFound` or `ErrSegmentNotFound` or something else... it feels like the service should be returning a better error.

**Pull request merged**